code/datums/action.dm

```diff
@@ -515,26 +513,28 @@
 		return spell.can_cast(owner)
 	return FALSE
 
-/datum/action/spell_action/UpdateButtonIcon()
-	if(button && !(. = ..()))
-		var/obj/effect/proc_holder/spell/S = target
-		if(!istype(S))
-			return
-		var/progress = S.get_availability_percentage()
-		var/col_val_high = 72 * progress + 128
-		var/col_val_low = 200 * progress
-		button.maptext = "<div style=\"font-size:6pt;color:[recharge_text_color];font:'Small Fonts';text-align:center;\" valign=\"bottom\">[round_down(progress * 100)]%</div>"
-		button.color = rgb(col_val_high, col_val_low, col_val_low, col_val_high)
-	else
-		button.maptext = null
+/datum/action/spell_action/apply_unavailable_effect()
+	var/obj/effect/proc_holder/spell/S = target
+	if(!istype(S))
+		return ..()
+	var/progress = S.get_availability_percentage()
+	if(progress == 1)
+		return ..() // This means that the spell is charged but unavailable due to something else
+
+	var/alpha = 220 - 140 * progress
+
+	var/image/img = image('icons/mob/screen_white.dmi', icon_state = "template")
+	img.alpha = alpha
+	img.appearance_flags = RESET_COLOR | RESET_ALPHA
+	img.color = "#000000"
+	img.plane = FLOAT_PLANE + 1
+	button.add_overlay(img)
+	// Make a holder for the charge text
+	var/image/count_down_holder = image('icons/effects/effects.dmi', icon_state = "nothing")
+	count_down_holder.plane = FLOAT_PLANE + 1.1
```

Comment:
```suggestion
	count_down_holder.plane = FLOAT_PLANE + 2
```
From the docs:
```
Possible values:
-10000 to 10000 (integers only), FLOAT_PLANE+p
````
First of all, I have no idea how this works with a float (does it get treated as 1 or as 2?..), but even if it works as you'd expect, you still have 20k values to work with, no need to be stingy here.

Reply:
I used it here to ensure it was just above the other image generated above. If say an admin or the game itself wants to apply a film for all players to see on a higher plane then they can more easily do it. A `FLOAT_PLANE + 2` will ensure that it would override this text atleast. They would most likely still use a higher value though